broadinstitute/single_cell_portal_core
# Enable remote CSFV bypass, omit .gz check for RDS files (SCP-5821)

This refines validation for Seurat data, and lets admins disable client-side validation for files uploaded via bucket path.

### Impact
Previously, Single Cell Portal did not account for [a convention in the R community](https://broadinstitute.slack.com/archives/CBEHTH601/p1727467481224789?thread_ts=1727444817.198009&cid=CBEHTH601) to omit .gz extensions from gzip-compressed RDS files.  Incidentally, this revealed a need to allow staff to occasionally disable client-side validation (CSFV) for remote files.

Now, RDS files can be gzipped or not, and won't throw any gzip-related error message.  And admins can now disable CSFV for remote files, like they already can for local files.

### Demo
Here's how it looks.

_Enable remote CSFV bypass_

https://github.com/user-attachments/assets/c16332d1-b86a-491a-b34d-f142fae48acc

_Omit .gz check for RDS files_

https://github.com/user-attachments/assets/47a6ff9a-c7da-4b1a-8c59-fb905b537c15

### Test
An automated test verifies omitting the .gz check for RDS files.  Optionally, to manually test:
1.  Set "clientside_validation" feature flag to "false"
2.  Go to AnnData upload UI for a study with no files
3.  Click "Use bucket path"
4.  Paste "invalid_disease_id.h5ad" into input
5.  Confirm no error message (if you get warning file doesn't exist, upload to bucket from `test/test_data/anndata`)
6.  Set "clientside_validation" feature flag to "true"
7.  Upload to bucket this file: `test/test_data/validation/missing_gz_extension.rds`
8.  Confirm no error message

This satisfies SCP-5821.
